### ISSUE TITLE
Icons are added in hidden places in desktop.

### ISSUE THREAD

**Issue by MarioMey** [mate-desktop/mate-desktop#571]
I have 2 monitors, side by side, but one lower than the other. You can see them here

![imagen](https://github.com/mate-desktop/mate-desktop/assets/4954109/3f0959a0-7aea-4497-bdf8-79e41833de2d)

The problem I'm facing now (it never happened to me before) is that the new files added to desktop, appear below the lowest icon at the left side. In other words: they are in a hidden place of the desktop.

In the screenshot above, there are new icons added. They are there, but they are... HIDDEN. In next screenshot, I click and drag to select everything below the visible screen. This selects one visible icon and everything in the selected area.

![imagen](https://github.com/mate-desktop/mate-desktop/assets/4954109/2675c604-86a5-441a-a1c6-eec249902fd4)

Then, I click and drag the visible icon and when I release it in other place, the hidden icons are finaly shown.

![imagen](https://github.com/mate-desktop/mate-desktop/assets/4954109/ad341694-ff9f-4781-8292-a43a560bbc0d)

I don't remember that this happened to me before... and I have this monitors configuration for some years.

#### Expected behaviour
New icons should not be added in hidden places in desktop.

#### Actual behaviour
New icons are added in hidden places in desktop.

#### Steps to reproduce the behaviour
Do what I explain avobe.

#### MATE general version
1.26.0-2

#### Package version
1.26.0-2

#### Linux Distribution
Debian 12

#### Link to bugreport of your Distribution (requirement)
If required, I'll do it. Just tell me.

**Comment by lukefromdc** [comment]
Since GTK3.22 there is only one `screen` that is a combination of all the monitors. As GTK has evolved, multimonitor handing has changed. I think GTK4 removed screens altogether, using just the `display` and` monitor` variables as what used to be done with screens had been moved to per-monitor solutions that are also recommended with GTK3. Tjhe original code dates back to the beginning of GTK2 as far as I know. Multi-monitor handling was quite different in those days.

 Looks like when caja is getting the workspace area, screen must be getting defined as the width and height of a rectangle draw to just contain the two rectangles defined by the monitors.

Getting out of this would require defining per-monitor workareas, which would also allow icons on multiple monitors in wayland where current git master puts icons on the FIRST monitor but renders the desktop background on each monitor

**Comment by GDGooD** [comment]
Duplicate of https://github.com/mate-desktop/mate-desktop/issues/411

**Comment by MarioMey** [comment]
> Duplicate of #411

Yes, it's the same problem. My report was added before yours. So, yours is duplicate of mine 😁.

**Comment by phu9Suop** [comment]
this hits me too,
